### PR TITLE
Refactor gateway messages

### PR DESCRIPTION
**Pull request by tkz00**
⚠️ Waiting on: [Backend PR](https://github.com/lambdaclass/mirra_backend/pull/473)

## Motivation

The messages for the battle history combined the information about an skill hitting a unit and the effect it had on the target (damage, heal) we realized this was flawed since it didn't allow for the correct feedback of these events in the client.

## Summary of changes

A new protobuf message for the battle history has been added, "Execution Received" which communicates stat changes, such as taking damage, and the "Effect Hit" has been left just for communicating an ability hitting a champion, which will match with the attack animations in the client, like a sword slash or a projectile hitting it's target, while the new "Execution Received" message matches with the indicators popping up and showing, for example, the damage taken.

## How has this been tested?

Let a battle play out as normal, everything should be the same, the damage numbers should popup from units and their healthbars should change when taking damage.

### CODE REVIEW

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ private void SetUpInitialState(Protobuf.Messages.BattleResult battleResult)
 	{
 		foreach (var unit in battleResult.InitialState.Units)
 		{
-			Debug.Log($"{unit.Id}, {unit.Health}, team: {unit.Team}");
+			// Debug.Log($"{unit.Id}, {unit.Health}, team: {unit.Team}");
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:74]:
```suggestion
```

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 2]
```diff
@@ -90,55 +90,30 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
 	{
 		foreach (var step in steps)
 		{
-			Debug.Log($"Step: {step.StepNumber}");
+			// Debug.Log($"Step: {step.StepNumber}");
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:93]:
```suggestion
```

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 3]
```diff
@@ -90,55 +90,30 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
...
 			yield return new WaitForSeconds(.3f);
 
-			foreach (var action in step.Actions)
+			foreach (var action in step.Actions.Where(action => action.ActionTypeCase != Protobuf.Messages.Action.ActionTypeOneofCase.SkillAction)
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:96]: Do you think extracting the collection of actions to a variable would be clearer? At least to give that collection a name and help the reader understand quickly what subset of actions is that.

**Review by ncontinanza — CHANGES_REQUESTED** [review]

File: client/Assets/Scripts/Battle/BattleManager.cs (hunk 1 above)
Comment (tkz00) [reply client/Assets/Scripts/Battle/BattleManager.cs]: Removed the comment

File: client/Assets/Scripts/Battle/BattleManager.cs (hunk 3 above)
Comment (tkz00) [reply client/Assets/Scripts/Battle/BattleManager.cs]: Yeah, I'm not sure about the name I used for the collection, ```actionsExcludingSkills```, but it's true it adds clarity.

**Review by ncontinanza — APPROVED** [review]

**Review by ncontinanza — APPROVED** [review]

**Review by martinacantaro — APPROVED** [review]
Tested it from the client and it works as intended

**Pull request merged**